pkg/insights/insightsclient/insightsclient.go

```diff
@@ -284,16 +284,67 @@ func (c *Client) Send(ctx context.Context, endpoint string, source Source) error
 
 // RecvReport perform a request to Insights Results Smart Proxy endpoint
 func (c Client) RecvReport(ctx context.Context, endpoint string) (*io.ReadCloser, error) {
+	resp, err := c.doGetRequest(ctx, endpoint, true)
+
+	if err != nil {
+		klog.Errorf("Unable to retrieve latest report for %s: %v", c.clusterVersion.Spec.ClusterID, err)
+		counterRequestRecvReport.WithLabelValues(c.metricsName, "0").Inc()
+		return nil, err
+	}
+
+	counterRequestRecvReport.WithLabelValues(c.metricsName, strconv.Itoa(resp.StatusCode)).Inc()
+	err = validateResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		return &resp.Body, nil
+	}
+
+	klog.Warningf("Report response status code: %d", resp.StatusCode)
+	return nil, fmt.Errorf("Report response status code: %d", resp.StatusCode)
```

Comment:
Nitpick: Couldn't we perhaps create the error, log it and then return it instead of two largely duplicate lines?

Reply:
Good catch. Thank you.